### DISCUSSION TITLE
UI not loading - Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'goto')

### DISCUSSION THREAD

**Discussion by GanizaniSitara** [open-webui/open-webui#931] [General]
# Bug Report

## Description

**Bug Summary:**
UI not loading staying with blank white screen.

**Steps to Reproduce:**
Install on Win10 standalone - Python 3.10.13, Node 18.16+

**Expected Behavior:**
Open WebUI starts.

**Actual Behavior:**
Blank screen. Error in Browser console 

`Uncaught (in promies) TypeError: Cannot read properties of undefined (reading 'goto')`

## Environment

- **Operating System:** Windows 10 
- **Browser (if applicable):** Chrome 120.0.6099.225, Edge 121.0.2277.112

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I have reviewed the troubleshooting.md document.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs. -- N/A

## Logs and Screenshots

**Browser Console Logs:**
![image](https://github.com/open-webui/open-webui/assets/7934938/96a45d29-53e1-4109-8772-75c7563e61f8)

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Manual installation. I have two machines one personal and one on corporate network. Identical configuration (Python versions, Node versions, Node package versions). Ollama validated running and reachable on both. 

Personal one works, corporate has the issue described here. Corporate has limited internet connectivity.  However transformers have been installed, tested. 

## Additional Information

Appears to be a **SvelteKit** promise issue as describe here 

https://stackoverflow.com/questions/72186150/why-do-i-get-error-uncaught-in-promise-typeerror-firstnamef-is-null

and here

https://svelte.dev/repl/2b95425c5aec4389bd484cf40a3fd1e5?version=3.48.0

We're failing on the second "await K("/auth")line here, I've never run uvicorn with non-minified debug, would be happy for a pointer to try and learn more. 

![image](https://github.com/open-webui/open-webui/assets/7934938/7ca30c94-e9ef-4e49-aa8c-ae0ed4e1b10c)

304's in console I don't see on the personal machine (all 200's there), but possibly unrelated(?).

![image](https://github.com/open-webui/open-webui/assets/7934938/6f793c33-d3f8-4ded-9841-c2a2280d3490)

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Unable to reproduce at all with node v20, if anyone manages to reliably reproduce the issue please let us know!

**Comment by GanizaniSitara** [comment]
Thank you, testing now repeated with v21.6.2, same result, will keep digging. Likely to be something to do with our corporate machine configuration, though am struggling to think what ...

**Comment by GanizaniSitara** [comment]
Found this to be some issue with esm-env package, the BROWSER import is either not set or evaluates to false. sounds like I'll need to do a simple build to prove this outside of the open-webui solution.
![image](https://github.com/open-webui/open-webui/assets/7934938/0057370d-78d7-485d-8b66-472e5e5e2cc0)

**Comment by GanizaniSitara** [comment]
Actually after further testing it seems that in my case it’s a build with source maps that resolves the issue so.

`npm run build —source-map`
